### PR TITLE
DM-1696: Remove Go Fish! landing page

### PR DESCRIPTION
**Pull request by emyl3**
### JIRA issue link
[DM-1696](https://agile6.atlassian.net/browse/DM-1696)

## Description - what does this code do?
This PR will remove the Go Fish! landing page.

## Testing done - how did you test it/steps on how can another person can test it 
1. Ensure you do not see the "Bulletin" drop down on the header
2. When navigating to `/competitions/go-fish` you are redirected to the home page

## Screenshots, Gifs, Videos from application (if applicable)
Screen Shot 2020-05-11 at 4 06 32 PM

## Acceptance criteria
- [X] Take down page
- [X] remove it from "Bulletin" dropdown
- [X] Archive it in some way (I think we do this anyway with the code base) in case we need to reuse it again (We can always recover it using `git` if needed.)

## Definition of done
- [ ] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating JIRA issue
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/controllers/competitions_controller.rb [hunk 1]
```diff
@@ -10,10 +10,7 @@ def shark_tank
   end
 
   def go_fish
-    @breadcrumbs = [
-        { text: 'Home', path: root_path },
-        { text: 'GoFish!' }
-    ]
+    redirect_to root_path
```
Comment (emyl3) [inline app/controllers/competitions_controller.rb:13]: ❓ Not sure if we need this... or if I can remove the `/competitions/go-fish` route entirely and a user who happens to navigate to `/competitions/go-fish` wouldn't get a weird error

File: app/views/shared/_header.html.erb [hunk 2]
```diff
@@ -58,15 +58,12 @@
   <%# </div> %>
   <nav class="usa-nav grid-col-fill flex-fill nav-container">
       <ul class="usa-nav__primary usa-accordion flex-justify-end">
-          <li class="usa-nav__primary-item margin-right-305">
+          <li class="usa-nav__primary-item margin-right-305 hidden">
```
Comment (emyl3) [inline app/views/shared/_header.html.erb:61]: ⚠️ hiding for now so this can be revived once the shark tank page is ready

**Comment by bradjohnson92008** [comment]
Looks good Elisa - but I think you are correct in that there will be a merge conflict when adding dm-1692 on top of this story in Master.

**Comment by ahay-agile6** [comment]
@bradjohnson92008 yeah, there will be conflicts.
let's merge this in and then pull from `master`

**Pull request merged**